core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -884,6 +895,17 @@ private Builder(TableMetadata base) {
       this.previousFileLocation = base.metadataFileLocation;
       this.previousFiles = base.previousFiles;
       this.refs = Maps.newHashMap(base.refs);
+      this.statistics =
+          base.statistics.stream()
+              .collect(
+                  Collectors.toMap(
+                      StatisticsFile::snapshotId,
+                      Function.identity(),
+                      (a, b) -> {
+                        throw new IllegalArgumentException(
+                            "Two statistics files for same snapshot");
```

Comment:
Is it a good idea to reject duplicate stats files? That seems like a way to needlessly break a table.

Reply:
good point